aws/data_source_aws_wafv2_ip_set.go

```diff
@@ -79,20 +81,23 @@ func dataSourceAwsWafv2IPSetRead(d *schema.ResourceData, meta interface{}) error
 	}
 
 	resp, err := conn.GetIPSet(&wafv2.GetIPSetInput{
-		Id:    aws.String(*foundIpSet.Id),
-		Name:  aws.String(*foundIpSet.Name),
+		Id:    foundIpSet.Id,
+		Name:  foundIpSet.Name,
 		Scope: aws.String(d.Get("scope").(string)),
 	})
 
-	if err != nil {
+	if err != nil || resp == nil || resp.IPSet == nil {
```

Comment:
i think we shouldn't lose the error here in case, so can we keep it as `if err != nil { return fmt.Errorf("message: %s", err) }`  and then the following lines can do the checks similar to `if resp == nil || resp.IPSet == nil { return fmt.Errorf("") }?

Reply:
I'm not sure how much extra value these specific nil checks add, that's why I grouped them.
Will change it tomorrow in all PRs.

Reply:
grouping the later 2 nil checks make sense 👍 i think the majority of the value comes from the initial `err != nil` and the error we'd get back to give the user context.